lib/git.rb

```diff
@@ -31,13 +33,50 @@ def log(path = nil, revstring = Constants::HEAD, options = {})
       logs.addPath(path) if path
       logs.setMaxCount(options[:max_count]) if options[:max_count]
       logs.setSkip(options[:skip]) if options[:skip]
+      jcommits = logs.call
       commits = Array.new
-      logs.call.each do |jcommit|
+      jcommits.each do |jcommit|
         commits << Commit.new(jrepo, jcommit)
       end
+      
+      if path && options[:follow]
+        jcommits = @jgit.log.add(ref).addPath(path).call
+        renamed = follow_renames(jcommits, path)
+        unless renamed.empty?
+          renamed.each do |jcommit|
+            commits << Commit.new(jrepo, jcommit)
+          end
+        end
```

Comment:
This `unless` block could be replaced by `commits += follow_renames(jcommits, path)` if `follow_renames` returned an array of `RJGit::Commit`s rather than `jcommits`.